frontend/repositories/example/apiDocumentRepository.ts

```diff
@@ -54,9 +58,13 @@ export class APIExampleRepository implements ExampleRepository {
   }
 
   async listStates(projectId: string, exampleId: number): Promise<ExampleStateItemList> {
-    const url = `/projects/${projectId}/examples/${exampleId}/states`
-    const response = await this.request.get(url)
-    return plainToInstance(ExampleStateItemList, response.data)
+    let result = Promise.resolve({count: 0, next: null, prev: null, items: []})
+    if (String(projectId) !== "undefined" && String(exampleId) !== "undefined") {
+      const url = `/projects/${projectId}/examples/${exampleId}/states`
+      const response = await this.request.get(url)
+      result = response.data
+    }
```

Comment:
For some reason this line gives my an error on my machine, I have to change it to:
```
result = Promise.resolve({count: response.data.count, next: response.data.next, prev: response.data.previous, items: response.data.results})
```
I had to map those params explicitly in other methods as well, where there is `result = response.data`. But probably it is just a quirk on my windows machine. I sometimes also still got 404 errors from other endpoints: `http://localhost:3000/v1/projects/undefined/scale-types` and `http://localhost:3000/v1/projects/undefined/examples/601/spans` but they are quite rare.